python/tests/test_data_array.py

```diff
@@ -95,6 +100,13 @@ def test_labels():
         da.labels
 
 
+def test_name():
+    a = sc.DataArray(data=1.0 * sc.units.m)
+    assert a.name == ''
+    a.name = 'abc'
+    assert a.name == 'abc'
+
+
```

Comment:
I think there should also be some I/O tests, at least a round trip write a DataArray, and then read it back from disk and check that it's equal to the original.